calculators/election_totals.rb

```diff
@@ -80,6 +89,17 @@ def fetch
             }
           end
         end
+
+        # Sum contribution by type information for whole election
+        [:supporting_type, :opposing_type].each do |type|
+          type_array = referendum.calculation(type)
+          unless type_array.nil?
+            type_array.each { |element|
+              cbt = {element['type'] => element['amount']}
+              by_type = by_type.merge(cbt){|key, oldval, newval| oldval + newval}
+            }
```

Comment:
a pattern we use elsewhere would be to replace lines 97-100 with:

```ruby
type_array.each do |type|
  by_type[type['type']] ||= 0
  by_type[type['type']] += type['amount']
end
```

which I think is a lot easier to understand.

Reply:
when I learn a new method I try to use it everywhere! :)